DesignSafe-CI/portal
# task/DES-2200: Return user projects from Jupyterhub JWT service account

## Overview: ##

Extended Juptyerhub sessions have had an issue where the jhub client token for project mounts is expired. CIC has suggested honoring requests coming from a TBD jupyterhub service account, that will carry a url param with the user's name. This service account will have a long-live token (managed by CIC), and so should always be able to get a user's project paths to mount.

## PR Status: ##


## Related Jira tickets: ##

* [DES-2200](https://jira.tacc.utexas.edu/browse/DES-2200)

## Testing Steps: ##
1.  add `AGAVE_JWT_SERVICE_ACCOUNT=jupyterh` in `designsafe.env`
2. Ensure you have ngrok running locally, and copy the forwarding url to `WEBHOOK_POST_URL` in `designsafe/settings/external_resource_secrets.py`
3. Submit a curl request with [this JWT](https://stache.utexas.edu/entry/9f64ae5f31ff3512665cd9bbf6b7392c) to your local projects api endpoint
    - e.g. `curl -i -H "X-JWT-ASSERTION-DESIGNSAFE: $JWT_TEST" "https://1234.ngrok.io/api/projects/?user=sal"`
4. Confirm the response is a valid request and returns a listing of your projects
